src/tablestore.rs

```diff
@@ -213,17 +205,21 @@ impl TableStore {
                 _ => SlateDBError::ObjectStoreError(e),
             })?;
 
-        self.cache_filter(*id, encoded_sst.filter);
+        self.cache_filter(*id, encoded_sst.info.filter_offset, encoded_sst.filter)
+            .await;
         Ok(SsTableHandle {
             id: *id,
             info: encoded_sst.info,
         })
     }
 
-    fn cache_filter(&self, sst: SsTableId, filter: Option<Arc<BloomFilter>>) {
-        {
-            let mut wguard = self.filter_cache.write();
-            wguard.insert(sst, filter);
+    async fn cache_filter(&self, sst: SsTableId, id: u64, filter: Option<Arc<BloomFilter>>) {
+        if let Some(cache) = &self.block_cache {
+            if let Some(filter) = filter {
+                cache
+                    .insert((sst, id as usize), CachedBlock::Filter(filter))
```

Comment:
now that we're using the offset as the id for everything, shall we just use a u64 as the key type?